CMakeLists.txt

```diff
@@ -6,6 +6,9 @@ option(DISABLE_PKGCONFIG "Disable PkgConfig module generation" OFF)
 option(DISABLE_CMAKE_CONFIG "Disable CMake package config module generation" OFF)
 option(ENABLE_WIN32_IO "Build the Windows I/O helper class" OFF)
 
+set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
```

Comment:
I think it's better to use the [VISIBILITY_INLINES_HIDDEN](https://cmake.org/cmake/help/latest/prop_tgt/VISIBILITY_INLINES_HIDDEN.html) (with the correct syntax) so it doesn't force this value on whatever other project is also compiled in the build tree.

When enabling that I also get the [CMP0063 warning](https://cmake.org/cmake/help/latest/policy/CMP0063.html). So we should probably bump the cmake requirement to 3.3.

For the same reason we should probably use the [CXX_VISIBILITY_PRESET](https://cmake.org/cmake/help/v3.17/prop_tgt/LANG_VISIBILITY_PRESET.html) property on the project as well, rather than setting it globally.

So something like this:
```
set_target_properties(ebml PROPERTIES
  CXX_VISIBILITY_PRESET hidden
  VISIBILITY_INLINES_HIDDEN ON
)
```

With gcc it generates something like this:
```
#ifdef EBML_STATIC_DEFINE
#  define EBML_DLL_API
#  define EBML_NO_EXPORT
#else
#  ifndef EBML_DLL_API
#    ifdef ebml_EXPORTS
        /* We are building this library */
#      define EBML_DLL_API __attribute__((visibility("default")))
#    else
        /* We are using this library */
#      define EBML_DLL_API __attribute__((visibility("default")))
#    endif
#  endif

#  ifndef EBML_NO_EXPORT
#    define EBML_NO_EXPORT __attribute__((visibility("hidden")))
#  endif
#endif
```
And calls gcc with `-fvisibility=hidden -fvisibility-inlines-hidden`. That seems right.

Reply:
Hi, good suggestion, do you want to author it ? :)